### PR TITLE
Add examples to tempdir

### PR DESCRIPTION
**Pull request by jbcden**
This still needs some work, but this adds some examples and aims to standardize some of the text more.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -38,11 +38,31 @@ const NUM_RETRIES: u32 = 1 << 31;
 const NUM_RAND_CHARS: usize = 12;
 
 impl TempDir {
-    /// Attempts to make a temporary directory inside of `tmpdir` whose name
+    /// Creates a `TempDir` inside of `tmpdir` whose name
```
Comment (killercup) [inline src/lib.rs:41]: Any reason you changed the wording from 'attempts to create' to 'creates'? As it returns a `Result`, this can actually fail.

Comment (jbcden) [inline src/lib.rs:41]: I changed it to include an `Errors` section which I think is a more standard way to discuss errors. I think it is nicer to have the main function description describe the happy path and have a section dedicated to what can go wrong.

**Comment by GuillaumeGomez** [comment]
I think it will conflict with #11.

**Comment by jbcden** [comment]
ah... I somehow totally missed that there was already a PR 😞

**Comment by jbcden** [comment]
Closing in favor of https://github.com/rust-lang-nursery/tempdir/pull/11

**Pull request closed without merging**